nemo/core/neural_graph.py

```diff
@@ -962,3 +967,79 @@ def unfreeze(self, module_names: Optional[List[str]] = None):
             if module.type == ModuleType.trainable:
                 # Unfreeze weights of the module.
                 module.unfreeze()
+            else:
+                logging.debug("Module `{}` is not trainable so cannot be unfrozen".format(name))
+
+    def save_to(self, filename: str, module_names: Optional[List[str]] = None):
+        """
+            Saves the state of trainable modules in the graph to a checkpoint file.
+            Args:
+                filename (string): Name of the file where the checkpoint will be saved.
+                module_names: List of modules to be frozen (Optional). If passed, all modules will be saved.
+            Raises:
+                KeyError: If name of the module won't be recognized.
+        """
+        # Work on all modules.
+        if module_names is None:
+            module_names = self._modules.keys()
+
+        # Prepare the "graph checkpoint".
+        chkpt = {"header": {}, "modules": {}}
+        chkpt["header"] = {"nemo_core_version": nemo_version, "name": self.name}
```

Comment:
This can be inlined above to avoid empty dictionary allocation.

Reply:
ok. But not sure how important that is...